sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/ClientSideRequestStatistics.java

```diff
@@ -304,11 +309,13 @@ public void serialize(
 
             try {
                 SystemInformation systemInformation = new SystemInformation();
-                long totalMemory = Runtime.getRuntime().totalMemory() / 1024;
-                long freeMemory = Runtime.getRuntime().freeMemory() / 1024;
-                long maxMemory = Runtime.getRuntime().maxMemory() / 1024;
+                Runtime runtime = Runtime.getRuntime();
+                long totalMemory = runtime.totalMemory() / 1024;
+                long freeMemory = runtime.freeMemory() / 1024;
+                long maxMemory = runtime.maxMemory() / 1024;
                 systemInformation.usedMemory = totalMemory - freeMemory + " KB";
                 systemInformation.availableMemory = (maxMemory - (totalMemory - freeMemory)) + " KB";
+                systemInformation.availableProcessors = runtime.availableProcessors();
```

Comment:
the number of CPU cores is not going to change per request.

For the static info please move to `DiagnosticsClientContext`. where other static info is kept

Reply:
Hmm - especially in Containerized COmpute it could actually change - because Cores available is a dynamic setting, correct?

Reply:
UBS in this case has a setting where they allow 0.5 -2 Cores in their AKS config

Reply:
To be honest - not sure what 0.5 would ever mean - maybe in HyperThreading.

Reply:
interesting. thanks for the explanation